core/src/gl/texture.cpp

```diff
@@ -45,6 +45,19 @@ Texture::Texture(const std::string& _file, TextureOptions _options, bool _genera
     stbi_image_free(pixels);
 }
 
+Texture::Texture(const unsigned char* data, size_t dataSize, TextureOptions options, bool generateMipmaps)
+    : Texture(0u, 0u, options, generateMipmaps) {
+    unsigned char* pixels;
+    int width, height, comp;
+
+    pixels = stbi_load_from_memory(data, dataSize, &width, &height, &comp, STBI_rgb_alpha);
```

Comment:
maybe you can make stbi code a separate function and call it from both the parameterized constructors you have for Texture.